central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -354,6 +357,67 @@ func (d *datastoreImpl) GetProfilesNames(ctx context.Context, q *v1.Query) ([]st
 	return profileNames, err
 }
 
+// GetProfilesNames gets the list of all distinct profile names which were configured either via StackRox or directly
+// in Compliance Operator for the query.
+func (d *datastoreImpl) GetProfilesNames(ctx context.Context, q *v1.Query) ([]string, error) {
+	var err error
+	q, err = withSACFilter(ctx, resources.Compliance, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// get all profiles configured in StackRox compliance schedules
+	resultProfileNames, err := d.getScanConfigProfileNames(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// get all profiles configured via Compliance Operator, not managed by StackRox
+	scanProfileNames, err := d.GetProfilesNamesFromOperatorScanSettings(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// merge profile name lists
+	for _, scanProfileName := range scanProfileNames {
+		if !slices.Contains[[]string, string](resultProfileNames, scanProfileName) {
+			resultProfileNames = append(resultProfileNames, scanProfileName)
+		}
+	}
+
+	return resultProfileNames, err
+}
+
+func (d *datastoreImpl) getScanConfigProfileNames(ctx context.Context, q *v1.Query) ([]string, error) {
+	clonedQuery := q.Clone()
+	// Build the select and group by on distinct profile name
+	clonedQuery.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.ComplianceOperatorConfigProfileName).Distinct().Proto(),
+	}
+	clonedQuery.GroupBy = &v1.QueryGroupBy{
+		Fields: []string{
+			search.ComplianceOperatorConfigProfileName.String(),
+		},
+	}
+
+	clonedQuery.Pagination = q.GetPagination()
+
+	var results []*distinctProfileName
+	results, err := pgSearch.RunSelectRequestForSchema[distinctProfileName](ctx, d.db, schema.ComplianceOperatorScanConfigurationV2Schema, clonedQuery)
+	if err != nil {
+		return nil, err
+	}
+	if len(results) == 0 {
+		return nil, nil
+	}
+
+	resultProfileNames := make([]string, 0, len(results))
+	for _, result := range results {
+		resultProfileNames = append(resultProfileNames, result.ProfileName)
+	}
+	return resultProfileNames, nil
```

Comment:
In this approach pagination is not supported, both query results are added here.